### PR TITLE
feature(views): Allow mapping views dirs via views.php files

### LINKED ISSUE #8546: master: prefix paths in views.php
Let's say we allow this in views.php:
```php
  'prefix/' => __DIR__ . '/vendors/prefix/dist',
```
A few problems:
1. We still have to scan `dist` to find view files. We no longer guess filenames at runtime (a good thing)
2. If `dist` has subdirs, we could pull a lot more into views than we want.

Would it compromise this feature to limit scanning to the given directory? I.e. you might have to do this:
```php
  'prefix/'        => __DIR__ . '/vendors/prefix/dist',
  'prefix/images/' => __DIR__ . '/vendors/prefix/dist/images',
```

@ewinslow

### LINKED ISSUE #8550: master: After #8546, improve /mod/files/views.php

### PR DESCRIPTION
**Pull request by mrclay**
views.php files can now map entire directory trees into the view system with a desired prefix for the view name.

Fixes #8546

chore(views): Restore _graphics for better back compatibility
This uses the new views.php prefix to map these to the same views.
- [x] Move core `_graphics` back into place
- [x] Add `Fixes #8550`
- [x] Update installer and docs references to graphics in views system

### CODE REVIEW

**Comment by ewinslow** [comment]
Yea, I suppose _graphics is just too easy an example, eh...? So will that just be:

```php
return [
  "default" => [
    "/" => __DIR__ . "/_graphics",
  ]
];
```

**Comment by mrclay** [comment]
Yep, done.

**Comment by ewinslow** [comment]
You can revert the htaccess changes also.

**Comment by mrclay** [comment]
Done

**Comment by mrclay** [comment]
This is an awesome feature.

**Comment by ewinslow** [comment]
Glad it's so compelling!

Soo... here's a thought: I'm working on composer compatibility for Elgg. If Elgg is installed as a composer dependency, that would move `_graphics` out of root path by necessity, so we'd need to handle `_graphics` access via elgg routing or `.htaccess` or something anyways.

Does that negate the value of putting _graphics back?

File: mod/twitter_api/views.php [hunk 1]
```diff
@@ -2,8 +2,6 @@
 
 return [
 	'default' => [
-		'twitter_api/sign-in-with-twitter-d.png' => __DIR__ . '/graphics/sign-in-with-twitter-d.png',
-		'twitter_api/sign-in-with-twitter-l.png' => __DIR__ . '/graphics/sign-in-with-twitter-l.png',
-		'twitter_api/sign-in-with-twitter.gif' => __DIR__ . '/graphics/sign-in-with-twitter.gif',
+		'twitter_api/' => __DIR__ . '/graphics',
```
Comment (ewinslow) [inline mod/twitter_api/views.php]: Wow, at this rate mapping directories seems like it will be more common than mapping files...?

**Comment by ewinslow** [comment]
installer and docs need to be updated as well because they reference views/default/favicon.ico

**Comment by ewinslow** [comment]
Also views/default/elgg_logo.png...

:( Sorry for this tedium.

**Comment by mrclay** [comment]
Strawman: move to new location (say, engine/graphics), tell site owners if they need filesystem BC they can symlink `_graphics` to the new location.

**Comment by ewinslow** [comment]
Would love to pull this in. Forget about the composer concerns. I'll deal with them in that PR.

Can we iron out the last blockers (the things that are causing Travis to fail).

**Comment by mrclay** [comment]
Should I do this? https://github.com/Elgg/Elgg/pull/8553#issuecomment-114092079

**Comment by ewinslow** [comment]
Nah

File: docs/conf.py [hunk 2]
```diff
@@ -138,7 +138,7 @@ def setup(app):
 # The name of an image file (within the static path) to use as favicon of the
 # docs.  This file should be a Windows icon file (.ico) being 16x16 or 32x32
 # pixels large.
-html_favicon = '../views/default/favicon.ico'
+html_favicon = '../_graphics/favicon.ico'
```
Comment (mrclay) [inline docs/conf.py]: @ewinslow is this path right?

File: install/config/htaccess.dist [hunk 3]
```diff
@@ -127,8 +127,6 @@ RewriteEngine on
 # hide all dot files/dirs (.git)
 RewriteRule (^\.|/\.) - [F]
 
-RewriteRule ^_graphics/(.*)$ cache/0/default/$1 [L]
```
Comment (mrclay) [inline install/config/htaccess.dist]: Should we really leave this in?

File: docs/conf.py (hunk 2 above)
Comment (ewinslow) [inline docs/conf.py]: Yes

File: install/config/htaccess.dist (hunk 3 above)
Comment (ewinslow) [inline install/config/htaccess.dist]: No, not for this PR. I can re-add if/when I need it for composer support.

**Comment by ewinslow** [comment]
Super cool! Thanks, Steve. I think I will like this feature a lot.

**Pull request merged**